moviepy/video/tools/drawing.py

```diff
@@ -162,20 +162,20 @@ def color_gradient(size,p1,p2=None,vector=None, r=None, col1=0,col2=1.0,
     elif shape == 'radial':
         if r is None:
             r = norm
-        if r==0:
+        elif r == 0:
```

Comment:
I don't think that this change is correct. The first `if` sets `r`. The subsequent `if` and `else` set `arr`. You could put an extra new line in-between the 2 `if`s to help distinguish them apart instead.